src/jltypes.c

```diff
@@ -728,7 +728,7 @@ static jl_value_t *intersect_typevar(jl_tvar_t *a, jl_value_t *b,
                 return both;
             }
             if (!jl_is_typevar(both))
-                both = (jl_value_t*)jl_new_typevar(underscore_sym, jl_bottom_type, both);
+                both = (jl_value_t*)jl_new_typevar(underscore_sym, jl_bottom_type, both);  // does this need a root? See call to jl_full_type in type_eqv_
```

Comment:
not this result (since `extend` is non-allocating when `both` is a tvar. but the result of `meet_tvars` did need a root before this new_typevar allocation (since it could be the result of `jl_type_intersection(a->ub, b->ub)`)